Investigate and fix system QA test test_integrity_sync

Hi team,


Some tests cases are failing in the QA system test `test_integrity_sync` in the **`master`** branch.

The `test_shared_files` and `test_extra_files` cases are failing. I have seen that the system.py tool has the old `ossec` and `ossec` **user** and **group** in one of its functions:

https://github.com/wazuh/wazuh-qa/blob/659f5161bc47f773d6380cea464c46189184bf8d/deps/wazuh_testing/wazuh_testing/tools/system.py#L38-L48

I have changed `ossec` to `wazuh` and the tests are still failing. It doesn't appear to be a synchronization problem as I have tested it manually and the `merged.mg` file and new groups folders are being synchronized properly. 

The following is the `test_integrity_sync` test output with the change `ossec`-`wazuh` already done.

We must investigate this error and fix the test or cluster functions that may be failing.

```
pytest test_integrity_sync/
================================================================================== test session starts ===================================================================================
platform linux -- Python 3.9.2, pytest-4.5.0, py-1.10.0, pluggy-0.13.1
rootdir: /home/manuel/git/wazuh-qa/tests/system/test_cluster
plugins: metadata-1.11.0, html-2.0.1, testinfra-5.0.0
collected 4 items                                                                                                                                                                        

test_integrity_sync/test_integrity_sync.py .FF.                                                                                                                                    [100%]

======================================================================================== FAILURES ========================================================================================
___________________________________________________________________________________ test_shared_files ____________________________________________________________________________________

    def test_shared_files():
        """Check if the content of each file is the same in all nodes.
    
        Update the content of the files in the master node and check if they are updated in the workers.
        Then, update the content in the workers and check if it is overwritten by the one in the master.
        """
        # Modify the content of each file in the master node to check if it is updated in the workers.
        for file in files_to_sync:
            host_manager.modify_file_content(host=test_hosts[0], path=file, content='test_content_from_master')
    
        time.sleep(time_to_sync)
    
        # Check whether files are correctly updated in the workers.
        for host in worker_hosts:
            for file in files_to_sync:
                result = host_manager.run_command(host, f'cat {file}')
>               assert result == 'test_content_from_master', f'File {file} inside {host} should contain ' \
                                                             f'"test_content_from_master", but it has: {result}'
E               AssertionError: File /var/ossec/etc/shared/test_group/merged.mg inside wazuh-worker1 should contain "test_content_from_master", but it has: #test_group
E                 !228 ar.conf
E                 restart-ossec0 - restart-ossec.sh - 0
E                 restart-ossec0 - restart-ossec.cmd - 0
E                 restart-wazuh0 - restart-ossec.sh - 0
E                 restart-wazuh0 - restart-ossec.cmd - 0
E                 restart-wazuh0 - restart-wazuh - 0
E                 restart-wazuh0 - restart-wazuh.exe - 0
E                 !0 test_file
E               assert '#test_group\...n!0 test_file' == 'test_content_from_master'
E                 + test_content_from_master
E                 - #test_group
E                 - !228 ar.conf
E                 - restart-ossec0 - restart-ossec.sh - 0
E                 - restart-ossec0 - restart-ossec.cmd - 0
E                 - restart-wazuh0 - restart-ossec.sh - 0
E                 - restart-wazuh0 - restart-ossec.cmd - 0...
E                 
E                 ...Full output truncated (4 lines hidden), use '-vv' to show

test_integrity_sync/test_integrity_sync.py:103: AssertionError
____________________________________________________________________________________ test_extra_files ____________________________________________________________________________________

clean_files = None

    def test_extra_files(clean_files):
        """Check if extra files in the workers are correctly deleted.
    
        Create files in the worker nodes inside directories which are not marked as "extra" in the cluster.json file.
        Only valid files, except for extra_valid ones, are those created in the master. Therefore, all the files created
        in the workers should be deleted.
        """
        # Create all specified files inside the worker nodes.
        for host in worker_hosts:
            for subdir in directories_to_create:
                host_manager.run_command(test_hosts[0], f'mkdir {subdir}')
                host_manager.run_command(test_hosts[0], f'chown wazuh:wazuh {subdir}')
            for file in files_to_sync:
                host_manager.run_command(host, f'touch {file}')
                host_manager.run_command(test_hosts[0], f'chown wazuh:wazuh {file}')
    
        time.sleep(time_to_sync)
    
        # Check if the files created before have been removed in the workers, as those were not extra_valid files.
        for host in worker_hosts:
            for file in files_to_sync:
                result = host_manager.run_command(host, f'ls {file}')
>               assert result == '', f"File {file} was expected to be removed from {host}, but it still exists."
E               AssertionError: File /var/ossec/etc/shared/test_group/merged.mg was expected to be removed from wazuh-worker1, but it still exists.
E               assert '/var/ossec/e...oup/merged.mg' == ''
E                 - /var/ossec/etc/shared/test_group/merged.mg

test_integrity_sync/test_integrity_sync.py:143: AssertionError
========================================================================== 2 failed, 2 passed in 407.51 seconds ==========================================================================
```

Regards,
Manuel.

--- a/deps/wazuh_testing/wazuh_testing/tools/system.py
+++ b/deps/wazuh_testing/wazuh_testing/tools/system.py
@@ -44,7 +44,7 @@ class HostManager:
         dest_path (str): Destination path
         check (bool, optional): Ansible check mode("Dry Run")(https://docs.ansible.com/ansible/latest/user_guide/playbooks_checkmode.html), by default it is enabled so no changes will be applied. Default `False`
         """
-        self.get_host(host).ansible("copy", f"src={src_path} dest={dest_path} owner=ossec group=ossec mode=0775",
+        self.get_host(host).ansible("copy", f"src={src_path} dest={dest_path} owner=wazuh group=wazuh mode=0775",
                                     check=check)
 
     def add_block_to_file(self, host: str, path: str, replace: str, before: str, after, check: bool = False):

--- a/tests/system/test_cluster/test_integrity_sync/test_integrity_sync.py
+++ b/tests/system/test_cluster/test_integrity_sync/test_integrity_sync.py
@@ -3,10 +3,9 @@
 # This program is free software; you can redistribute it and/or modify it under the terms of GPLv2
 
 import os
+import pytest
 import time
 from secrets import token_hex
-
-import pytest
 from wazuh_testing.tools import WAZUH_PATH
 from wazuh_testing.tools.system import HostManager
 
@@ -26,9 +25,10 @@ directories_to_create = [os.path.join(WAZUH_PATH, "etc", "shared", "test_group")
 # Files that, after created in the master, should be present in all nodes.
 files_to_sync = [os.path.join(WAZUH_PATH, "etc", "lists", "test_file"),
                  os.path.join(WAZUH_PATH, "etc", "rules", "test_file"),
-                 os.path.join(WAZUH_PATH, "etc", "decoders", "test_file"),
-                 os.path.join(directories_to_create[0], 'merged.mg'),
-                 os.path.join(directories_to_create[1], 'merged.mg')]
+                 os.path.join(WAZUH_PATH, "etc", "decoders", "test_file")]
+# merged.mg files that must be created after creating each group folder.
+merged_mg_files = [os.path.join(directories_to_create[0], "merged.mg"),
+                   os.path.join(directories_to_create[1], "merged.mg")]
 # Files inside directories where not 'all' files have to be synchronized, according to cluster.json.
 files_not_to_sync = [os.path.join(WAZUH_PATH, "etc", "test_file"),
                      os.path.join(WAZUH_PATH, "etc", "lists", 'ar.conf'),
@@ -38,6 +38,8 @@ files_not_to_sync = [os.path.join(WAZUH_PATH, "etc", "test_file"),
                      os.path.join(WAZUH_PATH, "etc", "lists", 'test.swp'),
                      os.path.join(directories_to_create[0], 'test_file'),
                      os.path.join(directories_to_create[1], 'test_file')]
+agent_conf_files_to_create = [os.path.join(directories_to_create[0], 'agent.conf'),
+                              os.path.join(directories_to_create[1], 'agent.conf')]
 
 
 @pytest.fixture(scope='function')
@@ -63,7 +65,7 @@ def test_missing_file(clean_files):
         host_manager.run_command(test_hosts[0], f'chown wazuh:wazuh {subdir}')
 
     # Create all specified files inside the master node.
-    for file in files_to_sync + files_not_to_sync:
+    for file in files_to_sync + files_not_to_sync + agent_conf_files_to_create:
         host_manager.run_command(test_hosts[0], f'touch {file}')
         host_manager.run_command(test_hosts[0], f'chown wazuh:wazuh {file}')
 
@@ -79,7 +81,7 @@ def test_missing_file(clean_files):
             assert perm == '660', f"{file} permissions were expected to be '660' in {host}, but they are {perm}."
         # Check that files which should not be synchronized are not sent to the workers. For example, only
         # merged.mg file inside /var/ossec/etc/shared/ directory should be synchronized, but nothing else.
-        for file in files_not_to_sync:
+        for file in files_not_to_sync + agent_conf_files_to_create:
             result = host_manager.run_command(host, f'ls {file}')
             assert result == '', f"File {file} was expected not to be copied in {host}, but it was."
 
@@ -90,23 +92,41 @@ def test_shared_files():
     Update the content of the files in the master node and check if they are updated in the workers.
     Then, update the content in the workers and check if it is overwritten by the one in the master.
     """
+    agent_conf_content = '<agent_config></agent_config>'
+    file_test_content_master = 'test_content_from_master'
+    file_test_content_worker = 'test_content_from_worker'
+
     # Modify the content of each file in the master node to check if it is updated in the workers.
     for file in files_to_sync:
-        host_manager.modify_file_content(host=test_hosts[0], path=file, content='test_content_from_master')
+        host_manager.modify_file_content(host=test_hosts[0], path=file, content=file_test_content_master)
+
+    # Modify the content of agent.conf files to check if each merged.mg file is updated in master and synchronized in
+    # workers.
+    for file in agent_conf_files_to_create:
+        host_manager.modify_file_content(host=test_hosts[0], path=file,
+                                         content=f"{agent_conf_content}\n")
 
     time.sleep(time_to_sync)
 
-    # Check whether files are correctly updated in the workers.
+    # Check whether files are correctly updated in the workers or not.
     for host in worker_hosts:
         for file in files_to_sync:
             result = host_manager.run_command(host, f'cat {file}')
-            assert result == 'test_content_from_master', f'File {file} inside {host} should contain ' \
-                                                         f'"test_content_from_master", but it has: {result}'
+            assert result == file_test_content_master, f'File {file} inside {host} should contain ' \
+                                                       f'{file_test_content_master}, but it has: {result}'
+
+    # Check whether merged.mg files are correctly updated in master and synchronized in workers or not.
+    for host in test_hosts:
+        for file in merged_mg_files:
+            result = host_manager.run_command(host, f'cat {file}')
+            # The agent.conf content will be before the !0 test_file line in merged.mg.
+            assert agent_conf_content in result, f'File {file} inside {host} should contain ' \
+                                                 f'{agent_conf_content}, but it has: {result} '
 
     # Update the content of files in the worker node.
     for host in worker_hosts:
         for file in files_to_sync:
-            host_manager.modify_file_content(host=host, path=file, content='test_content_from_worker')
+            host_manager.modify_file_content(host=host, path=file, content=file_test_content_worker)
 
     time.sleep(time_to_sync)
 
@@ -114,8 +134,8 @@ def test_shared_files():
     for host in worker_hosts:
         for file in files_to_sync:
             result = host_manager.run_command(host, f'cat {file}')
-            assert result == 'test_content_from_master', f'File {file} inside {host} should contain ' \
-                                                         f'"test_content_from_master", but it has: {result}'
+            assert result == file_test_content_master, f'File {file} inside {host} should contain ' \
+                                                       f'{file_test_content_master}, but it has: {result}'
 
 
 def test_extra_files(clean_files):

--- a/tests/system/test_cluster/test_integrity_sync/test_integrity_sync.py
+++ b/tests/system/test_cluster/test_integrity_sync/test_integrity_sync.py
@@ -22,13 +22,13 @@ client_keys_path = os.path.join(WAZUH_PATH, "etc", "client.keys")
 # Subdirectories to be synchronized.
 directories_to_create = [os.path.join(WAZUH_PATH, "etc", "shared", "test_group"),
                          os.path.join(WAZUH_PATH, "var", "multigroups", "test_dir")]
-# Files that, after created in the master, should be present in all nodes.
+
+# Files that, after created in the master node, should be present in all nodes.
 files_to_sync = [os.path.join(WAZUH_PATH, "etc", "lists", "test_file"),
                  os.path.join(WAZUH_PATH, "etc", "rules", "test_file"),
-                 os.path.join(WAZUH_PATH, "etc", "decoders", "test_file")]
-# merged.mg files that must be created after creating each group folder.
-merged_mg_files = [os.path.join(directories_to_create[0], "merged.mg"),
-                   os.path.join(directories_to_create[1], "merged.mg")]
+                 os.path.join(WAZUH_PATH, "etc", "decoders", "test_file"),
+                 os.path.join(directories_to_create[1], 'merged.mg')]
+
 # Files inside directories where not 'all' files have to be synchronized, according to cluster.json.
 files_not_to_sync = [os.path.join(WAZUH_PATH, "etc", "test_file"),
                      os.path.join(WAZUH_PATH, "etc", "lists", 'ar.conf'),
@@ -38,8 +38,10 @@ files_not_to_sync = [os.path.join(WAZUH_PATH, "etc", "test_file"),
                      os.path.join(WAZUH_PATH, "etc", "lists", 'test.swp'),
                      os.path.join(directories_to_create[0], 'test_file'),
                      os.path.join(directories_to_create[1], 'test_file')]
-agent_conf_files_to_create = [os.path.join(directories_to_create[0], 'agent.conf'),
-                              os.path.join(directories_to_create[1], 'agent.conf')]
+
+# merged.mg and agent.conf files that must be created after creating a group folder.
+merged_mg_file = os.path.join(directories_to_create[0], "merged.mg")
+agent_conf_file = os.path.join(directories_to_create[0], 'agent.conf')
 
 
 @pytest.fixture(scope='function')
@@ -65,7 +67,7 @@ def test_missing_file(clean_files):
         host_manager.run_command(test_hosts[0], f'chown wazuh:wazuh {subdir}')
 
     # Create all specified files inside the master node.
-    for file in files_to_sync + files_not_to_sync + agent_conf_files_to_create:
+    for file in files_to_sync + files_not_to_sync + [agent_conf_file]:
         host_manager.run_command(test_hosts[0], f'touch {file}')
         host_manager.run_command(test_hosts[0], f'chown wazuh:wazuh {file}')
 
@@ -81,7 +83,7 @@ def test_missing_file(clean_files):
             assert perm == '660', f"{file} permissions were expected to be '660' in {host}, but they are {perm}."
         # Check that files which should not be synchronized are not sent to the workers. For example, only
         # merged.mg file inside /var/ossec/etc/shared/ directory should be synchronized, but nothing else.
-        for file in files_not_to_sync + agent_conf_files_to_create:
+        for file in files_not_to_sync + [agent_conf_file]:
             result = host_manager.run_command(host, f'ls {file}')
             assert result == '', f"File {file} was expected not to be copied in {host}, but it was."
 
@@ -100,11 +102,10 @@ def test_shared_files():
     for file in files_to_sync:
         host_manager.modify_file_content(host=test_hosts[0], path=file, content=file_test_content_master)
 
-    # Modify the content of agent.conf files to check if each merged.mg file is updated in master and synchronized in
+    # Modify the content of the agent.conf file to check if merged.mg file is updated in master and synchronized in
     # workers.
-    for file in agent_conf_files_to_create:
-        host_manager.modify_file_content(host=test_hosts[0], path=file,
-                                         content=f"{agent_conf_content}\n")
+    host_manager.modify_file_content(host=test_hosts[0], path=agent_conf_file,
+                                     content=f"{agent_conf_content}\n")
 
     time.sleep(time_to_sync)
 
@@ -115,13 +116,12 @@ def test_shared_files():
             assert result == file_test_content_master, f'File {file} inside {host} should contain ' \
                                                        f'{file_test_content_master}, but it has: {result}'
 
-    # Check whether merged.mg files are correctly updated in master and synchronized in workers or not.
+    # Check whether the merged.mg file is correctly updated in master and synchronized in workers or not.
     for host in test_hosts:
-        for file in merged_mg_files:
-            result = host_manager.run_command(host, f'cat {file}')
-            # The agent.conf content will be before the !0 test_file line in merged.mg.
-            assert agent_conf_content in result, f'File {file} inside {host} should contain ' \
-                                                 f'{agent_conf_content}, but it has: {result} '
+        result = host_manager.run_command(host, f'cat {merged_mg_file}')
+        # The agent.conf content will be before the !0 test_file line in merged.mg.
+        assert agent_conf_content in result, f'File {merged_mg_file} inside {host} should contain ' \
+                                             f'{agent_conf_content}, but it has: {result} '
 
     # Update the content of files in the worker node.
     for host in worker_hosts:

